### ISSUE TITLE
Duplicating worksheet is not copying styling

### ISSUE THREAD

**Issue by vce-wabbott** [PHPOffice/PhpSpreadsheet#2494]
This is:

```
- [X] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?

Expecting a worksheet to be fully duplicated using this code

$clonedWorksheet = clone $spreadsheet->getSheetByName('Worksheet 1');
$clonedWorksheet->setTitle('Copy of Worksheet 1');
$spreadsheet->addSheet($clonedWorksheet);

### What is the current behavior?

Some cell styles like fills and font colors are not copied over

### What are the steps to reproduce?

Please provide a [Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php
	$reader = new \PhpOffice\PhpSpreadsheet\Reader\Xlsx();
	$spreadsheet = $reader->load('Elevation.xlsx');

	$new_elevation = clone $spreadsheet->getActiveSheet();
	$new_elevation->setTitle('Max Potential');
	$spreadsheet->addSheet($new_elevation);

	$spreadsheet->setActiveSheetIndexByName('Max Potential');

	$writer = new \PhpOffice\PhpSpreadsheet\Writer\Xlsx($spreadsheet);
	$writer->save('php://output');
```

### Which versions of PhpSpreadsheet and PHP are affected?

1.21
[Elevation.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/7842554/Elevation.xlsx)

**Comment by oleibman** [comment]
Actually, most of the styling (borders, bold, etc.) is being handled just fine. But, you are right - there are at least 2 problems. First, we not yet support the default Excel styling of hyperlinks. This is more difficult than it seems.

The more important problem is the fill colors. That seems to be due to PR #1694. That change was discussed at length in issue #2366. It was felt at the time that the regression cases were due to user coding errors. However, your sample file makes it clear that such a problem can make it into the wild. What is happening is that the Xlsx reader is expecting the fill color to be specified as, say:
```xml
<fgColor rgb="FFBFBFBF"/>
```
In other words, it is expecting an 8-character string despite the attribute name of `rgb`. However, your file specifies:
```xml
<fgColor rgb="BFBFBF" />
```
The Style Reader tries to set ARGB using this 6-character string, and it fails.

I will add a fix for this to PR #2471, which makes extensive changes to the Xlsx Style Reader, and which will remain in draft status for at least a couple of weeks. If you see additional styling differences apart from the 2 discussed above, please let me know.

**Comment by vce-wabbott** [comment]
This Excel output was generated by the open source library EPPlus - this isn't the first time I've seen issues as a result of how EPPlus outputs its XLSX files. I'll see if I can find other stylings that aren't being copied over properly.
